### DISCUSSION TITLE
error message in compiling STM32F030C8T6> RBOARD_32F030C8T6 >variant.h error

### DISCUSSION THREAD

**Discussion by GitHubedklam** [r-pv/rSTM32F030C8T6_ProMiniPlus#4] [Q&A] [answered]
hello; I am new to using STM32F030C8T6 with arduino IDE.  I followed r-pv instruction (r-pv/rSTM32F030C8T6_ProMiniPlus) . When I compile a sample sketch "blink" I received the following error > 
  "  Warning: Board STM32:stm32:Nu030F4cleo_64 doesn't define a 'build.board' preference. Auto-set to: STM32_NU030F4CLEO_64
c:/users/edward/appdata/local/arduino15/packages/stm32/tools/xpack-arm-none-eabi-gcc/9.2.1-1.1/bin/../lib/gcc/arm-none-eabi/9.2.1/../../../../arm-none-eabi/bin/ld.exe: cannot open linker script file C:\Users\Edward\AppData\Local\Arduino15\packages\STM32\hardware\stm32\1.9.0\variants\RBOARD_F030C8T6/ldscript.ld: Invalid argument
collect2.exe: error: ld returned 1 exit status
exit status 1
Error compiling for board Generic STM32F0 series. "

Please help me to resolve this.Thank in advance.

**Comment by r-pv** [comment]
Hi, do you select the board **RBOARD_F030C8T6** on the arduino IDE? selecting "generic stm32f0 series" on the "board" menu, then pick "RBOARD_F030C8T6" on the "board part number" menu.

The error sounds like the board it's not selected and/or the editing of the board.txt file failed.

**Reply by GitHubedklam** [reply]
I checked and I am sending you clip shots so you can double check to see what I missed.  The arduino ide under  "generic stm32f0 series" on the "board" menu > no  RBOARD_F030C8T6 . instead it has STM32F030C8T6 STProMiniPlus  so I choose this as it matches the board. During compile the first warning almost immediately " Warning: Board STM32:stm32:Nu030F4cleo_64 doesn't define a 'build.board' preference. Auto-set to: STM32_NU030F4CLEO_64" it this normal ? finally failure with warning as noted earlier. 
Screen caputures below and thanks> 
[Screen clipping taken 1142021 251 PM.pdf](https://github.com/r-pv/rSTM32F030C8T6_ProMiniPlus/files/5816750/Screen.clipping.taken.1142021.251.PM.pdf)
[Screen clipping taken 1142021 239 PM.pdf](https://github.com/r-pv/rSTM32F030C8T6_ProMiniPlus/files/5816751/Screen.clipping.taken.1142021.239.PM.pdf)
[Screen clipping taken 1142021 240 PM.pdf](https://github.com/r-pv/rSTM32F030C8T6_ProMiniPlus/files/5816752/Screen.clipping.taken.1142021.240.PM.pdf)
[Screen clipping taken 1142021 244 PM.pdf](https://github.com/r-pv/rSTM32F030C8T6_ProMiniPlus/files/5816753/Screen.clipping.taken.1142021.244.PM.pdf)

**Comment by r-pv** [comment, marked as answer]
Hi, yes, the last screen shows a boards.txt really mess up, it shows like lines were added at the end of other lines instead of just add the code at the end of one block of text.

Your boards.txt looks like:

```
line0line4
line1line5
line2line6
line3line7

line8
line9
...
```

and, per instructions, should look like:

```
line0
line1
line2
line3

line4
line5
line6
line7

line8
line9
...
```
so the idea it's that you'll find this text:

```
# DEMO_F030F4 board
# Support: Serial1 (USART1 on PA3, PA2)
GenF0.menu.pnum.DEMO_F030F4=STM32F030F4 Demo board
GenF0.menu.pnum.DEMO_F030F4.upload.maximum_data_size=4096
GenF0.menu.pnum.DEMO_F030F4.upload.maximum_size=16384
GenF0.menu.pnum.DEMO_F030F4.build.mcu=cortex-m0
GenF0.menu.pnum.DEMO_F030F4.build.board=DEMO_F030F4
GenF0.menu.pnum.DEMO_F030F4.build.series=STM32F0xx
GenF0.menu.pnum.DEMO_F030F4.build.product_line=STM32F030x6
GenF0.menu.pnum.DEMO_F030F4.build.variant=DEMO_F030F4
GenF0.menu.pnum.DEMO_F030F4.build.cmsis_lib_gcc=arm_cortexM0l_math
```
and just add the text on the instructions to leave it as:

```
# DEMO_F030F4 board
# Support: Serial1 (USART1 on PA3, PA2)
GenF0.menu.pnum.DEMO_F030F4=STM32F030F4 Demo board
GenF0.menu.pnum.DEMO_F030F4.upload.maximum_data_size=4096
GenF0.menu.pnum.DEMO_F030F4.upload.maximum_size=16384
GenF0.menu.pnum.DEMO_F030F4.build.mcu=cortex-m0
GenF0.menu.pnum.DEMO_F030F4.build.board=DEMO_F030F4
GenF0.menu.pnum.DEMO_F030F4.build.series=STM32F0xx
GenF0.menu.pnum.DEMO_F030F4.build.product_line=STM32F030x6
GenF0.menu.pnum.DEMO_F030F4.build.variant=DEMO_F030F4
GenF0.menu.pnum.DEMO_F030F4.build.cmsis_lib_gcc=arm_cortexM0l_math

# RBOARD_F030C8T6 board
# Support: Serial1 (USART1 on PA3, PA2)
GenF0.menu.pnum.RBOARD_F030C8T6=STM32F030C8 STProMiniPlus
GenF0.menu.pnum.RBOARD_F030C8T6.upload.maximum_data_size=8192
GenF0.menu.pnum.RBOARD_F030C8T6.upload.maximum_size=65536
GenF0.menu.pnum.RBOARD_F030C8T6.build.mcu=cortex-m0
GenF0.menu.pnum.RBOARD_F030C8T6.build.board=RBOARD_F030C8T6
GenF0.menu.pnum.RBOARD_F030C8T6.build.series=STM32F0xx
GenF0.menu.pnum.RBOARD_F030C8T6.build.product_line=STM32F030x8
GenF0.menu.pnum.RBOARD_F030C8T6.build.variant=RBOARD_F030C8T6
GenF0.menu.pnum.RBOARD_F030C8T6.build.cmsis_lib_gcc=arm_cortexM0l_math
```

Did you make a copy of the boards.txt to start again? if not I'm afraid that you'll have to reinstall arduino to start again.

**Reply by GitHubedklam** [reply]
I did have make a backup.  I made the change as per your instruction. I selected "STM32F030C8T6 STProMiniPlus" in "Board part number".
I compiled and upload.  It uploaded ok and my test program " Blink" works.  :)   
Many thanks for your help and saved me many frustration. I was about to trash this (bought by mistake) and buy the STM32F103C which I wanted at the first place.

**Comment by r-pv** [comment]
Glad that you make it work. 
I like this MCU, when I got them was like 5pieces x$3usd, the bare IC of course, isn't as powerfull as the F103C but it's cheap and still it's way super-charged than, lets say, an atmega328p.

BTW, the ST Core v2.0 it seems that will have support for this MCU, so it won't be needed any modification.

**Reply by GitHubedklam** [reply]
thanks for the insight

**Reply by GitHubedklam** [reply]
Thanks for sharing and provide details. Your post is the only one I could find ( after searching for hours ! )on how to use this board F030C with Arduino IDE.
